architecture/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,457 @@
+                "dynamicTypeLoading": true
+            },
+            "extensions": {
+                "radius": "br:biceptypes.azurecr.io/radius:latest",
+                "aws": "br:biceptypes.azurecr.io/aws:latest"
+                "Mycompany": "br:biceptypes.azurecr.io/mycompany:latest"
+            }
+        }
+        ```
+    1. Create a recipe for the custom resource type Plaid
+
+        ```bash
+        rad recipe register plaid-dev –resource-type “Mycompany.Messaging/Plaid" --template-kind bicep --template-path "<myregistry>/plaid:latest"
+        ```
+    1. Create a app.bicep with plaid resource type
+        ```bicep
+        param plaidName string
+        param plaidDescription string
+        param plaidQueue string
+        resource plaid 'Mycompany.Messaging/Plaid@2024-08-09-preview' = {
+            name: plaidName
+            properties: {
+            recipe:{
+                name: 'plaid-dev'
+            }
+            }
+        }
+        ```
+    1. Deb uses the rad CLI to deploy the application with the custom resource type Plaid
+        
+        ```bash
+        rad deploy app.bicep 
+        ```
+    The application is deployed with the custom resource type Plaid in Radius. Deb can iterate on the changes, compile and test the application with the custom resource type.
+
+    
+1. Amy is a system integrator who wants to author a custom resource PostgreSQL in Radius
+
+    **Pre-requisites**: User has typespec compiler/CLI installed
+
+    1. Amy uses the rad CLI to scaffold a `typespec` template definition
+
+        ```bash
+        rad resource-provider init Application.Datastores --template-kind typespec
+        ```
+        Radius scaffolds a typespec project for the resource provider `Application.Datastores` with individual typespec files for the resource types. 
+
+    1. Amy authors a typespec schema for the PostgreSQL resource type in typespec format
+            
+        ```typespec
+            import Radius;
+
+            using Radius;
+            
+            namespace Applications.Datastores;
+                    
+            @doc("PostgreSQL database")
+            model PostgreSQLResource {
+            @doc("The name of the database")
+            name: string;
+            }
+            @doc("PostgreSQL properties")
+            model PostgreSQLProperties {
+                @doc("The host name of the target PostgreSQL database")
+                host?: string;
+                @doc("The port value of the target PostgreSQL database")
+                port?: int32;
+                @doc("The username for the PostgreSQL database")
+                username?: string;
+            }
+        ```
+
+    1. VSCode Type spec extension warns Amy on the errors and warnings as he types
+        
+        ![alt text](2024-06-resource-extensbility/errors.png)
+
+    1. Amy uses auto-completion for the schema and properties
+
+        ![alt text](2024-06-resource-extensbility/autocompletion.png)
```

Comment:
another broken link

Reply:
I do see the images rendered in markdown view... anything I am missing
![image](https://github.com/user-attachments/assets/e745c67d-9710-412f-8676-d62183d470db)